test/regress_dns.c

```diff
@@ -1679,10 +1679,12 @@ test_bufferevent_connect_hostname(void *arg)
 	tt_int_op(be_outcome[2].what, ==, !emfile ? BEV_EVENT_CONNECTED : BEV_EVENT_ERROR);
 	tt_int_op(be_outcome[2].dnserr, ==, 0);
 	tt_int_op(be_outcome[3].what, ==, !emfile ? BEV_EVENT_CONNECTED : BEV_EVENT_ERROR);
+	/*
+	 * Some platforms check for localhost explicitly, and therefore may succeed without opening any files *
+	 * e.g. https://github.com/openbsd/src/blob/53e0023678f73561cc0c0c07e49830be23d94673/lib/libc/asr/getaddrinfo_async.c#L234
+	 */
 	if (!emfile) {
 		tt_int_op(be_outcome[3].dnserr, ==, 0);
-	} else {
-		tt_int_op(be_outcome[3].dnserr, !=, 0);
```

Comment:
But then this test does not make sense anymore.
Maybe it will be better to explicitly skip it for openbsd?
Or/and I'm ok to run it only for linux

Reply:
It seems very "did this platform put the explicit check in their getaddrinfo" rather than a platform-specific set-up. I will add the check, though, if that makes sense to you.

Reply:
The initial problem was #600

So this was a real regression test, so I would prefer to keep it, let's run only for linux and done with it, this will be enough for now